### PR TITLE
update: add auto resize textarea

### PR DESCRIPTION
**Pull request by liuboaibc**
**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
add auto resize textarea

**Which issue(s) this PR fixes**:
Fixes [#496](https://github.com/kubesphere/console/issues/496)

### CODE REVIEW

File: src/components/Base/AutoResizeTextArea/index.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
```
Comment (leoendless) [inline src/components/Base/AutoResizeTextArea/index.jsx:1]: Better named as TextArea， with a bool property ``autoResize``.

File: src/components/Base/AutoResizeTextArea/index.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    placeholder: 'placeholder',
+    onChange: jest.fn(),
+  }
+  const wrapper = mount(<AutoResizeTextArea {...props} />)
+  const textarea = wrapper.find('textarea')
+
+  expect(textarea).toHaveClassName(styles.textarea)
+  expect(textarea).toHaveProp({ placeholder: props.placeholder })
+  expect(textarea).toHaveProp({ value: props.value })
+})
```
Comment (leoendless) [inline src/components/Base/AutoResizeTextArea/index.test.js:36]: Need to test if the textarea auto resized when mounted and input changed.

File: src/components/Base/AutoResizeTextArea/index.jsx (hunk 1 above)
Comment (liuboaibc) [reply src/components/Base/AutoResizeTextArea/index.jsx:1]: got it

File: src/components/Base/AutoResizeTextArea/index.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    placeholder: PropTypes.string,
+    value: PropTypes.string,
+    onChange: PropTypes.func,
+  }
+
+  static defaultProps = {
+    name: '',
+    placeholder: '',
+    value: '',
+    onChange() {},
```
Comment (leoendless) [inline src/components/Base/AutoResizeTextArea/index.jsx:38]: Need to give a default value to ``maxHeight``.

**Pull request closed without merging**